packages/entity-server/src/routes/hierarchy/middleware/attachEntityContext.ts

```diff
@@ -46,13 +50,16 @@ const validateEntitiesAndBuildContext = async (
     throwNoAccessError(entityCodes);
   }
 
-  if (!entities.every(entity => userCanAccessEntity(entity, allowedCountries))) {
+  const allowedEntities = entities.filter(entity =>
+    userCanAccessEntity(entity, allowedCountries, rootEntity),
+  );
```

Comment:
I was under the false assumption previously that we'd return only the entities with access if requesting for multiple entities. Turns out we'd thrown an error if any of the entities couldn't be accessed... I believe we wish to only return entities with access though... So I've made that change

Reply:
It sort of does... however my theory is that we're just making it more lenient here? Previously it would have thrown an error, and given that we don't see that error anywhere then we ought not be encountering this scenario anywhere in practice currently? 

However, all the same this seems like better behaviour to me...